src/wallet/spend.cpp

```diff
@@ -744,9 +756,17 @@ static bool CreateTransactionInternal(
     coin_selection_params.m_change_fee = coin_selection_params.m_effective_feerate.GetFee(coin_selection_params.change_output_size);
     coin_selection_params.m_cost_of_change = coin_selection_params.m_discard_feerate.GetFee(coin_selection_params.change_spend_size) + coin_selection_params.m_change_fee;
 
+    // We want to drop the change to fees if:
+    // 1. The change output would be dust
+    // 2. The change is within the (almost) exact match window, i.e. it is less than or equal to the cost of the change output (cost_of_change)
+    const auto dust = GetDustThreshold(change_prototype_txout, coin_selection_params.m_discard_feerate);
+    // TODO: disambiguate between min_change and m_cost_of_change
+    if (dust > coin_selection_params.m_cost_of_change) coin_selection_params.m_cost_of_change = dust;
```

Comment:
In 6423998409736be5728bb5826cfa89e1033ed816 "WIP. Return change from SelectionResult"

```suggestion
    coin_selection_params.m_cost_of_change = std::max(GetDustThreshold(change_prototype_txout, coin_selection_params.m_discard_feerate), coin_selection_params.m_cost_of_change);
```